### PR TITLE
Hydroponics tray changes and new plants

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
rscadd: You can now remove all plants and weeds from a hydroponics tray by using a spade on it.
rscadd: Added meatwheat, a mutated variant of wheat that can be crushed into meat substitute.
rscadd: Added ambrosia gaia, a mysterious plant that is rumored to provide nutrients and water to any soil - hydroponic or otherwise - that it's planted in.
rscadd: Added cherry bombs, a mutated variant of blue cherries that have an explosively good taste. Oh, and don't pluck the stems.
/ :cl:
- You can now remove all plants and weeds from a hydroponics tray by using a spade on it. This process takes five seconds.
- Hydroponics trays now have a unique variable, `using_irrigation`, that replaces the previous iteration of irrigation hoses that used the `anchored` variable to track it instead.
- Fixes a possible runtime error that can occur when using irrigation hoses to transfer non-water reagents (specifically nutrients, although it hasn't been tested with any other reagents).
- Added meatwheat, which is mutated from normal wheat plants. It's essentially a blood-red version of wheat that can be crushed into a meat slab by activating it with an empty hand. These meatwheat clumps are slightly bloodier than normal meat, but it still works for recipes that use normal meat slabs.
- Added ambrosia gaia, which is mutated from ambrosia deus. It's yellow, glows brightly, and contains only earthsblood (renamed adminordrazine). The plant is destroyed after a single harvest, and its product can be planted in an empty hydroponics tray to cause it to become self-sufficient, no longer requiring nutrients, water, or weed/pest removal. Gaia-blessed trays will glow bright yellow in comparison to normal trays, and gaia-blessed soil will turn yellow. Both will emit light.
- Added cherry bombs, which are mutated from blue cherries. They're orange and have a black stem, and start with a tiny amount of black powder in them. Activating them in an empty hand will ignite them and cause them to explode after three seconds, with explosion strength varying based on potency (at the most, it's around that of a minibomb).

_A normal tray and a gaia-blessed tray respectively_
trays

_Meatwheat, ambrosia gaia, and cherry bomb plants and seeds_
plants

**Token: #18218**

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Nice.

**Comment by Xhuis** [comment]
I might add some more plants that I've been toying around with in my head for... well, several months now.

**Comment by AdamElTablawy** [comment]
Meatwheat sounds lame. Bloodroot sounds better.

**Comment by Xhuis** [comment]
I chose meatwheat because it rhymes, but I can easily do a different name. I'll brainstorm some ideas.

**Comment by Xhuis** [comment]
I added ambrosia gaia, an elusive and miraculous variety of the famous plants. Read more in the OP.

File: code/modules/hydroponics/hydroponics.dm [hunk 1]
```diff
@@ -651,7 +671,23 @@
...
 	//Called when mob user "attacks" it with object O
-	if(istype(O, /obj/item/weapon/reagent_containers) )  // Syringe stuff (and other reagent containers now too)
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks/grown/ambrosia/gaia)) //Checked early on so it doesn't have to deal with composting checks
+		if(self_sustaining)
+			user << "<span class='warning'>This [name] is already self-sustaining!</span>"
+			return
+		if(myseed || weedlevel)
+			user << "<span class='warning'>[src] needs to be clear of plants and weeds!</span>"
+			return
+		if(alert(user, "This will [src] self-sustaining but consume [O] forever. Are you sure?", "[name]", "I'm Sure", "Abort") == "Abort")
```
Comment (Xhuis) [inline code/modules/hydroponics/hydroponics.dm]: To-do: fix grammar here

**Comment by Jalleo** [comment]
Very good idea. Cant wait for the station to be filled with hydro trays growing weed that dosent need topping up.

**Comment by RemieRichards** [comment]
@Chocobro

**Comment by Chocobro** [comment]
is there a way for players to get adminordrazine through the ambrosia gaia? ensure there's no way of converting it. pictures of the plants themselves would also be appreciated - other than that, all lookin good dude good shit 👌👀👌👀👌👀👌👀👌👀 good shit go౦ԁ sHit👌 thats ✔ some good👌👌shit right👌👌there👌👌👌 right✔there ✔✔if i do ƽaү so my self 💯 i say so 💯 thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ💯 👌👌 👌НO0ОଠOOOOOОଠଠOoooᵒᵒᵒᵒᵒᵒᵒᵒᵒ👌 👌👌 👌 💯 👌 👀 👀 👀 👌👌Good shit

(thx 4 ping remie)

**Comment by Chocobro** [comment]
HAHA SINCE WHEN DOES GITHUB HAVE THOSE EMOTES

**Comment by RemieRichards** [comment]
@Chocobro it can be ground in a beaker.
The beaker'd contents can then be put into a queen bee.
The queen bee can then produce a buttload of workers that all produce adminordrazine and then put it in their honeycomb
The honeycomb can then be ground for adminordrazine.

**Comment by Chocobro** [comment]
oh shoot
in that case you may want to make it a subtype that can't be ground or can't be mixed with bees
assuming it isn't that already it's 9am i am not looking at code

**Comment by Dorsisdwarf** [comment]
Make worker bees explode if they contain earthsblood, so you can make beeicide bombers with some good timing

**Comment by Chocobro** [comment]
i love it
someone promote this man

File: code/modules/hydroponics/hydroponics.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/hydroponics/hydroponics.dm]: Add sanity checks here too.

**Comment by Core0verload** [comment]
Admindrazine for players? What could possibly go wrong?

File: code/modules/hydroponics/grown/ambrosia.dm [hunk 2]
```diff
@@ -49,3 +49,25 @@
...
+//Ambrosia Gaia
+/obj/item/seeds/ambrosia/gaia
+	name = "pack of ambrosia gaia seeds"
+	desc = "These seeds grow into ambrosia gaia, incredibly rare but imbued with the power of a whole ecosystem."
+	icon_state = "seed-ambrosia_gaia"
+	species = "ambrosia_gaia"
+	plantname = "Ambrosia Gaia"
+	product = /obj/item/weapon/reagent_containers/food/snacks/grown/ambrosia/gaia
+	mutatelist = list()
+	reagents_add = list("earthsblood" = 0.4)
```
Comment (phil235) [inline code/modules/hydroponics/grown/ambrosia.dm]: No, players should never be able to get their hands on adminordrazine.

**Comment by Xhuis** [comment]
All right, all right. I'll replace the earthsblood with something else.

**Comment by Iamgoofball** [comment]
looks like piss

**Comment by Jalleo** [comment]
@Xhuis just make it a good but not oh god uts OP run away healing item

File: code/modules/hydroponics/grown/ambrosia.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/modules/hydroponics/grown/ambrosia.dm]: i'm gonna be honest here adminord is pretty tame nowadays compared to a competent chemist mix

**Comment by Xhuis** [comment]
@Jalleo That's the plan. I'm going to make it a custom reagent with stats similar to the nanites that Syndicate mediborgs get, but somewhat weaker and giving you the trademark ambrosia drugginess.

**Comment by Xhuis** [comment]
**New in this commit:**
* Earthsblood is now closer to restorative nanites from Syndicate mediborgs in effects, although it makes you druggy and overdose causes heavy toxin damage and hallucinations.
* Added cherry bombs. Mutated from blue cherries, they have a small amount of black powder in them and can be activated in an empty hand to pluck part of the stem off and ignite them. After three or so seconds, they'll explode with a power based on potency, with the largest possible explosion being a little weaker than a minibomb.

**Comment by Core0verload** [comment]
Make cherry bomb explosions based on heating reagents inside, not on potency. It would make it more compatible with botany LEGO genetics.

Making the whole "can be activated in an empty hand to pluck part of the stem off and ignite them" into a gene would add FUN to botany.

File: code/modules/hydroponics/grown/ambrosia.dm [hunk 3]
```diff
@@ -49,3 +49,25 @@
...
+/obj/item/seeds/ambrosia/gaia
+	name = "pack of ambrosia gaia seeds"
+	desc = "These seeds grow into ambrosia gaia, incredibly rare but imbued with breathtaking potential."
+	icon_state = "seed-ambrosia_gaia"
+	species = "ambrosia_gaia"
+	plantname = "Ambrosia Gaia"
+	product = /obj/item/weapon/reagent_containers/food/snacks/grown/ambrosia/gaia
+	mutatelist = list()
+	reagents_add = list("earthsblood" = 0.4, "nutriment" = 0.2, "vitamin" = 0.1)
+	rarity = 100 //These are some pretty good plants right here
```
Comment (optimumtact) [inline code/modules/hydroponics/grown/ambrosia.dm]: ﻿ｒａｒｅ  ｐｌａｎｔｓ

File: code/modules/reagents/chemistry/holder.dm [hunk 4]
```diff
@@ -128,7 +128,7 @@ var/const/INJECT = 5 //injection
 	return id
 
 /datum/reagents/proc/trans_to(obj/target, amount=1, multiplier=1, preserve_data=1, no_react = 0)//if preserve_data=0, the reagents data will be lost. Usefull if you use data for some strange stuff and don't want it to be transferred.
-	if(!target )
+	if(!target || !total_volume)
```
Comment (optimumtact) [inline code/modules/reagents/chemistry/holder.dm]: what what this for?

**Comment by Core0verload** [comment]
@Xhuis, if you think that `rarity = 100` makes plant mutation less common, you are wrong.

**Comment by Xhuis** [comment]
I know, but it's still a very powerful plant and Centcom would probably be interested in it, hence the point value/discovery value.

**Comment by Chocobro** [comment]
the gaia growth animation should be a slow pulse/glow imo. a strobe like that doesn't look right in my head.

it's merged now anyway but yeah

File: code/modules/reagents/chemistry/holder.dm (hunk 4 above)
Comment (phil235) [inline code/modules/reagents/chemistry/holder.dm]: You should've removed the same chec on line 137 `src.total_volume<=0`

**Comment by Iamgoofball** [comment]
@Xhuis please do what CO said with the genes and memes

**Pull request merged**